src/modules/enrollment/components/staffAssignment/StaffAssignmentCard.tsx

```diff
@@ -0,0 +1,113 @@
+import { Divider, Stack, Typography } from '@mui/material';
+import React, { useMemo, useState } from 'react';
+import {
+  CommonDetailGridContainer,
+  CommonDetailGridItem,
+} from '@/components/elements/CommonDetailGrid';
+import Loading from '@/components/elements/Loading';
+import TitleCard from '@/components/elements/TitleCard';
+import { useEnrollmentDashboardContext } from '@/components/pages/EnrollmentDashboard';
+import IconButtonContainer from '@/modules/enrollment/components/IconButtonContainer';
+import EditStaffAssignmentDialog from '@/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog';
+import {
+  RelationshipToHoH,
+  StaffAssignmentDetailsFragment,
+  useGetHouseholdStaffAssignmentsQuery,
+} from '@/types/gqlTypes';
+
+interface StaffAssignmentCardProps {
+  householdId: string;
+}
+
+const StaffAssignmentCard: React.FC<StaffAssignmentCardProps> = ({
+  householdId,
+}) => {
+  const { enrollment } = useEnrollmentDashboardContext();
+
+  const {
+    data: { household } = {},
+    loading,
+    error,
+  } = useGetHouseholdStaffAssignmentsQuery({
+    variables: { id: householdId },
+  }); // fixme - this query gets repeated twice with the same variables; why?
+
+  const assignmentsByType = useMemo(() => {
+    if (!household?.staffAssignments) return {};
+
+    return household.staffAssignments.nodes.reduce(
+      (
+        map: Record<string, StaffAssignmentDetailsFragment[]>,
+        a: StaffAssignmentDetailsFragment
+      ) => {
+        const l = map[a.staffAssignmentType];
+        map[a.staffAssignmentType] = l ? [...l, a] : [a];
+        return map;
+      },
+      {}
+    );
+  }, [household?.staffAssignments]);
+
+  const [editDialogOpen, setEditDialogOpen] = useState(false);
+
+  if (!enrollment) return;
+  if (error) throw error;
+
+  return (
+    <>
+      {household && editDialogOpen && (
+        <EditStaffAssignmentDialog
+          household={household}
+          open={editDialogOpen}
+          onClose={() => setEditDialogOpen(false)}
+        />
+      )}
+      <TitleCard
+        title='Staff Assignment'
+        actions={
+          enrollment.relationshipToHoH ===
+            RelationshipToHoH.SelfHeadOfHousehold &&
+          enrollment?.access?.canEditEnrollments && (
+            <IconButtonContainer
+              onClick={() => setEditDialogOpen(true)}
+              Icon={EditIcon}
+              ButtonProps={{ 'aria-label': `Edit Staff Assignment` }}
+            />
+          )
+        }
+      >
+        {loading && <Loading />}
+        {!loading && household?.staffAssignments && (
+          <>
+            <Divider />
+            {household.staffAssignments.nodesCount === 0 && (
+              <Typography sx={{ p: 2 }} color='text.secondary'>
+                No Staff Assigned
+              </Typography>
+            )}
+            <CommonDetailGridContainer>
+              {Object.entries(assignmentsByType).map(([type, assignments]) => (
```

Comment:
This seems fine for mvp but I have a couple of concerns
* it's not clear to me that the assignments will be ordered in a consistent way. Maybe they are?
* I'm not sure about the value of grouping assignments by role rather than just showing each staff assignment and role on their own row. Seems like there's some value in having the presentation be consistent
* One possibility would be to try and re-use our the "Current Staff Assignments" table from the modal, perhaps with a 'read-only' variant to hide the 'unassign' column

Reply:
Yeah, I was thinking about this as I was writing it too. Let's raise it in the design session tomorrow.